modules/tinymce/src/plugins/accordion/main/ts/core/Keyboard.ts

```diff
@@ -0,0 +1,64 @@
+import Editor from 'tinymce/core/api/Editor';
+import VK from 'tinymce/core/api/util/VK';
+
+import * as Utils from './Utils';
+
+const setupEnterKeyInSummary = (editor: Editor): void => {
+  editor.on('keydown', (event): void => {
+    if (event.shiftKey) {
+      return;
+    }
+    if (event.keyCode !== VK.ENTER) {
+      return;
+    }
+    if (!Utils.isInSummary(editor)) {
+      return;
+    }
+    event.preventDefault();
+    editor.execCommand('ToggleAccordion');
+  });
+};
+
+const setupEnterKeyInAccordionBody = (editor: Editor): void => {
+  editor.on('keydown', (event): void => {
+    if (event.shiftKey) {
+      return;
+    }
+    if (event.keyCode !== VK.ENTER) {
+      return;
+    }
+
+    const node = editor.selection.getNode();
+    if (!editor.dom.isEmpty(node)) {
```

Comment:
That's correct. I've updated the code to look for the nearest block parent, but it is not working correctly either. After applying formatting for the empty paragraph, this markup appears:

```html
<span id="_mce_caret" data-mce-bogus="1" data-mce-type="format-caret"><span style="text-decoration: underline;" data-mce-style="text-decoration: underline;">﻿</span></span>
```

And `editor.dom.isEmpty` returns `false` for this structure for some reason. Is it expected behavior or a bug in the `isEmpty` function?

Reply:
I wonder if that is a bug I think it's because it is filled with a zwnbsp character and that is treated as contents. I guess in theory we could look specifically for this element in the domUtils.isEmpty code and if it's just containing that zwnbsp character we treat that as nothing but I wonder if that would have other side effects.